Code/GraphMol/ROMol.cpp

```diff
@@ -459,10 +433,9 @@ unsigned int ROMol::addBond(Bond *bond_pin, bool takeOwnership) {
                                      bond_p->getEndAtomIdx(), d_graph);
   CHECK_INVARIANT(ok, "bond could not be added");
   d_graph[which] = bond_p;
+  bond_p->setIdx(numBonds);
   numBonds++;
-  //  int res = rdcast<int>(boost::num_edges(d_graph));
-  bond_p->setIdx(numBonds - 1);
-  return numBonds;  // res;
+  return numBonds;
```

Comment:
Might it be time we let the return here be the bond index?  I know this is an API change, but we should make it at some point.

Reply:
It would be sensible to do this at some point, but there's no way to do it in a way that maintains backwards compatibility. We'd have to think very carefully about how/when to do that